lint/rule_registry.go

```diff
@@ -32,33 +32,33 @@ func (r Rules) Copy() Rules {
 
 // Register registers the list of rules.
 // Return an error if any of the rules is found duplicate in the registry.
-func (r Rules) Register(rules ...Rule) error {
+func (r RuleRegistry) Register(rules ...ProtoRule) error {
 	for _, rl := range rules {
-		if !rl.Info().Name.IsValid() {
-			return fmt.Errorf("%q is not a valid RuleName", rl.Info().Name)
+		if !rl.GetName().IsValid() {
+			return fmt.Errorf("%q is not a valid RuleName", rl.GetName())
 		}
 
-		if _, found := r[rl.Info().Name]; found {
-			return fmt.Errorf("duplicate rule name %q", rl.Info().Name)
+		if _, found := r[rl.GetName()]; found {
+			return fmt.Errorf("duplicate rule name %q", rl.GetName())
 		}
 
-		r[rl.Info().Name] = rl
+		r[rl.GetName()] = rl
 	}
 	return nil
 }
 
 // All returns all rules.
-func (r Rules) All() []Rule {
-	rules := make([]Rule, 0, len(r))
+func (r RuleRegistry) All() []ProtoRule {
```

Comment:
Consider change `All` to `AllProtoRules` as well, if you stick with `RuleRegistry`.